src/bootstrap/bin/main.rs

```diff
@@ -16,17 +16,20 @@ fn main() {
     let mut build_lock;
     let _build_lock_guard;
     if cfg!(any(unix, windows)) {
+        // Get existing contents of lock file
         let path = config.out.join("lock");
+        let pid = t!(std::fs::read_to_string(&path));
```

Comment:
You need to check if this path exists before reading the file. if it doesn't exist, there's no process holding the lock and this will panic because the file won't be found.

(even better than checking if it exists is to handle `ErrorKind::NotFound`, to avoid TOCTOU issues.)